### DISCUSSION TITLE
Nuxt3 dev environment through proxy?

### DISCUSSION THREAD

**Discussion by Schetnan** [nuxt/nuxt#18190] [Questions]
I have run into what might be a showstopper scenario for us for Nuxt3.  I am wondering if there might be a solution to the problem.  Here is the scenario.

1). Our site consists of about 20 different nuxt2 applications that we tie together with a proxy in development.  The proxy does the following things.  
   A). Ties all of our applications running under different ports together under a common host - https://localhost:8443.  We have a file that maps each application path -ie. /my/app to its appropriate host - ie. http://localhost:3000
   B). Our context-root ( sort of ) is defined in each application via the router: { basePath: "" } attribute in nuxt.config.
   C). The router than pipes our application through a corporate proxy which injects a header and footer and corporate web component library javascript around our application.

This all works quite well in nuxt2.  The only drawback is that we have to refresh the browser in development after making a code change rather than instantaneous push to the browser.  It is not a big deal.

2). I tried doing the same thing with Nuxt3 thinking that it would work similarly.  And I ran into the following issue ( so far ).
   A). Even though the proxy routes paths to our app, the apps still want to communicate directly to the host to pull _nuxt files.  Since the app is proxied and running under a different host - https://localhost:8443, the browser can't find the files and returns 404 errors.
  B). This didn't seem to be a problem in Nuxt2 since the files needed to run the app were loaded up front.
  C).  This no longer seems to be the case with Nuxt3.  Nuxt3 seems to want to load a bunch of files from the root host after loading the app.  I initially discovered a problem when a simple two way binding through v-model on an input field was not working.  One way binding was working but not two way.
  D). I then loaded the application through http://localhost:3000 the non proxied host and two-way databinding worked find and the 404 errors went away. 
  E). If this doesn't work, it pretty much ruins our development environment and makes it difficult to maintain and debug our apps.

Question:  Is there a way already, might there be a way eventually, or could a way be developed to provide possibly one of the following options.

1). Have a dev server context root that could be configured so that all files - pages, etc. plus everything else needed for development _nuxt files etc. get served through the same context root?  I think this might help to deal with the proxy issue.

2). Have a setting, option to tell the nuxt development environment to be a little less smart and push everything that is needed to run the page to the browser all at once?

**Comment by Schetnan** [comment]
Looking at our nuxt2 apps, I noticed why it is working correctly with the proxy.  Given the following. nuxt.config.js config

```
  router: {
    base: '/fcp/csearch',
  }
```

So, proxying to our app via the following URL:  `https://localhost:8443/fcp/csearch`, ( The app actually running at `http://localhost:8100`), I can observe from the browser that resources are being loaded in the following manner.

```
https://localhost:8443/fcp/csearch/__webpack_hmr/client
https://localhost:8443/fcp/csearch/_nuxt/components/procedure-alert.js
```

The router's basePath is being appended to the front of every URL including component and _nuxt/ files etc.  This allows our app to working when being proxied to.

Observing nuxt3 with the following configuration:

```
router: {
  /fcp/myApp
}
```

I see the following from the browser.

```
myApp:1 Refused to apply style from 'https://localhost:8443/_nuxt/pages/index.vue?vue&type=style&index=0&scoped=true&lang.css' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.
myApp:527 GET https://localhost:8443/_nuxt/@vite/client net::ERR_ABORTED 404 (Not Found)
myApp:527 GET https://localhost:8443/_nuxt/entry.mjs net::ERR_ABORTED 404 (Not Found)
myApp:1 Refused to apply style from 'https://localhost:8443/_nuxt/pages/index.vue?vue&type=style&index=0&scoped=true&lang.css' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.
myApp:1 GET https://localhost:8443/_nuxt/@vite/client net::ERR_ABORTED 404 (Not Found)
myApp:1 GET https://localhost:8443/_nuxt/entry.mjs net::ERR_ABORTED 404 (Not Found)
```

It seems that nuxt3 is currently not attaching the router base path to some of the _nuxt files etc.  If it did, then router.base would affectively be the contextRoot allowing our app to work when proxied to.

**Comment by dbhagen** [comment]
Just curious, can you try adding this into the `nuxt.config.ts`?
```typescript
import { defineNuxtConfig } from 'nuxt3'

// https://v3.nuxtjs.org/docs/directory-structure/nuxt.config
export default defineNuxtConfig({
  vite: {
    server: {
      origin: 'https://localhost:8443/fcp/myApp'
    }
  }
})

**Reply by Schetnan** [reply]
Good thought.  I tried that and it doesn't seem to make a difference.
